src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs

```diff
@@ -151,6 +167,24 @@ static bool MethodHasCallbackAttributes(MethodInfo method)
         }
     }
 
+    private static bool HasAttribute(Type type, string[] attributeNames)
+    {
+        foreach (CustomAttributeData cattr in CustomAttributeData.GetCustomAttributes(type))
+        {
+            try
+            {
+                for (int i = 0; i < attributeNames.Length; ++i)
+                    if (cattr.AttributeType.FullName == attributeNames [i])
+                        return true;
+            }
+            catch
+            {
+                // Assembly not found, ignore
+            }
+        }
+        return false;
+    }
```

Comment:
Maybe:
```suggestion
    private static bool HasAttribute(MemberInfo element, params string[] attributeNames)
    {
        foreach (CustomAttributeData cattr in CustomAttributeData.GetCustomAttributes(element))
        {
            try
            {
                for (int i = 0; i < attributeNames.Length; ++i)
                {
                    if (cattr.AttributeType.FullName == attributeNames [i] ||
                        cattr.AttributeType.Name == attributeNames[i])
                    {
                        return true;
                    }
                }
            }
            catch
            {
                // Assembly not found, ignore
            }
        }
        return false;
    }
```

.. then `MethodHasCallbackAttributes` can be removed, and the call replaced with `HasAttribute(method, "System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute", "MonoPInvokeCallbackAttribute")`.

Reply:
This won't work since MonoPInvokeCallbackAttribute can be in any namespace.